react/react-hooks/why-are-react-hooks-required.md

```diff
@@ -0,0 +1,232 @@
+
+If you are new to Web Development or React in particular, you might have heard this buzz word "React Hooks" a lot and might be finding it overwhelming too. You may be having questions like – "Should I learn it? Is it important? Is it even required for a beginner like me?".
+
+While I cannot straight away answer your questions, what I can do is help you understand why do React Hooks exist and what motivated their intoduction to React by the React Team, so that you can decide for yourself whether to learn React Hooks or not.
+
+## Let's get started
+
+Just so that we don't get overwhelmed with all the things that hooks have to offer, let's first define our goal for this article and we'll stick to it.
+
+> In order to answer why are React Hooks required, we will need to understand one thing: What problems do React Hooks solve?
+
+## Background
+
+To get a little context, let's see how React has evolved over the years. Initially when React was release, there was no concept of class in JavaScript and hence it had to be handled by React. Back then, the React Components were created using `React.createClass` API as shown in the code snipped below.
+
+```javascript
+const ExampleComponent = React.createClass({
+  getInitialState () {
+    return {
+        // ...
+    }
+  },
+  componentDidMount () {
+    // setup
+    // add listener for feature 1
+    // add listener for feature 2
+    // and so on...
+  },
+  componentDidUpdate () {
+    // logic to update state
+  },
+  componentWillUnmount () {
+    // clean up
+    // remove listener for feature 1
+    // remove listener for feature 2
+    // and so on...
+  },
+
+  render() {
+    return (
+      // ...
+    )
+  }
+})
+```
+
+### Post ES6 release
+
+With release of ES6, `class` keyword was introduced in JavaScript and it was a new way to create classes in JavaScript. Later, React deprecated the `createClass` API and moved to `React.Component` API. Now we could create React Components using JavaScript native way of creating classes, as shown in the snippet below.
+
+```js
+class ExampleComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // ...
+    };
+
+    this.exampleFunction = this.exampleFunction.bind(this);
+  }
+  componentDidMount() {
+    // setup
+    // add listener for feature 1
+    // add listener for feature 2
+    // and so on...
+  }
+
+  componentDidUpdate() {
+    // logic to update state
+  }
+
+  componentWillUnmount () {
+    // clean up
+    // remove listener for feature 1
+    // remove listener for feature 2
+    // and so on...
+  },
+
+  exampleFunction() {
+    //...
+
+    this.setState({
+      // ...
+    });
+  }
+
+  render() {
+    // ...
+  }
+}
+```
+
+With this change, React got better than earlier, but it still had some tradeoffs. Everytime while creating a component, it was compulsory to use constructor and call `super(props)` and also `.bind` every function using `.this` keyword in the constructor.
```

Comment:
`React got better than earlier` and `React got better` both convey the same meaning. You can opt-out `than earlier`.